### PR TITLE
suppress expansion of unused raft spectral templates

### LINKED ISSUE #2678: spectral_clustering build is suspiciously slow and contributes to CI build timeouts
We need to look at why `spectral_clustering.cu` takes ~10x the time to compile vs. all others.  On my personal workstation I observed >6 minutes to build this file vs. <0.5 minutes for others.
We have been seeing several CI build timeouts lately for specific platforms, and the excessively long time to build spectral_clustering is believed to be a contributing factor.  Plus, local development from a clean build is impacted for every developer.

### PR DESCRIPTION
**Pull request by cjnolet**
Closes #2678 

raft updated some k-means logic and dramatically increased the compile time of our spectral clustering implementation.  This PR adds an include that will suppress the expansion of templates that we are not using.

### CODE REVIEW

**Comment by BradReesWork** [comment]
added labels just so CI will run

**Review by ChuckHastings — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Review by rlratzel — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

File: python/cugraph/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,22 @@ option(FIND_CUGRAPH_CPP "Search for existing CUGRAPH C++ installations before de
...
+  enable_language(CUDA)
+
+  set(CUGRAPH_VERSION_MAJOR "${cugraph-python_VERSION_MAJOR}")
+  set(CUGRAPH_VERSION_MINOR "${cugraph-python_VERSION_MINOR}")
+
+
+  # We need to call get_raft due to cuML asking for raft::nn and
+  # raft::distance targets
+  # see issue https://github.com/rapidsai/cuml/issues/4843
+  include(../../cpp/cmake/thirdparty/get_raft.cmake)
```
Comment (vyasr) [inline python/cugraph/CMakeLists.txt:53]: @cjnolet good find. The underlying issue here will be fixed by https://github.com/rapidsai/rapids-cmake/pull/154.

**Comment by vyasr** [comment]
For future reference the issue this PR fixes that caused #2707 to fail is that rapids-cmake's export commands don't support `COMPONENTS`, so the distance component requirement of libcugraph isn't being propagated to the pylibcugraph and cugraph packages. Once rapidsai/rapids-cmake#154 is merged we should be able to get rid of the extra `get_raft.cmake` includes.

**Pull request merged**